c10/metal/special_math.h

```diff
@@ -531,20 +531,20 @@ inline float xlog1py(T x, T y) {
 }
 
 template <typename T>
-inline float entr(T a) {
+inline T entr(T a) {
```

Comment:
Probably worth a bit more discussion, but perhaps those functions should always return `float`